### DISCUSSION TITLE
C-I/Tab behaviour is weird, is this expected?

### DISCUSSION THREAD

**Discussion by IndianBoy42** [neovim/neovim#28718] [Q&A] [answered]
`<C-v><C-i>` and `<C-v><Tab>` gives the same results: a tab space

Even `<C-v><C-m>` and `<C-v><CR>` gives the same results: `^M`

Yet given normal mode mappings for each separately they are distinguished properly. Ultimately the functionality is there but its confusing to try and debug if the terminal was setup properly.

But it gets weirder, if there is a buffer local mapping for C-i, and a global mapping for Tab: then C-i and Tab will both only call the buffer local mapping for C-i. If both mappings are global then they are distinguished, if both mappings are local then they are distinguished. But if one is buffer local then both go to that one.

**Comment by zeertzjq** [comment, marked as answer]
> `<C-v><C-i>` and `<C-v><Tab>` gives the same results: a tab space
> 
> Even `<C-v><C-m>` and `<C-v><CR>` gives the same results: `^M`

This is expected. See `:h i_CTRL-V`:
```
						*i_CTRL-V*
CTRL-V		Insert next non-digit literally.  It's also possible to enter
		the decimal, octal or hexadecimal value of a character
		|i_CTRL-V_digit|.
		The characters typed right after CTRL-V are not considered for
		mapping.
		For special keys, the CTRL modifier may be included into the
		key to produce a control character.  If there is no control
		character for the key then its |key-notation| is inserted.
		Note: When CTRL-V is mapped (e.g., to paste text) you can
		often use CTRL-Q instead |i_CTRL-Q|.

						*i_CTRL-Q*
CTRL-Q		Same as CTRL-V.
		Note: Some terminal connections may eat CTRL-Q, it doesn't
		work then.  It does work in the GUI.

CTRL-SHIFT-V				*i_CTRL-SHIFT-V* *i_CTRL-SHIFT-Q*
CTRL-SHIFT-Q	Works just like CTRL-V, but do not try to include the CTRL
		modifier into the key.
```

You can use `:inoremap <C-V> <C-S-V>` to achieve different results for these two keys.

> But it gets weirder, if there is a buffer local mapping for C-i, and a global mapping for Tab: then C-i and Tab will both only call the buffer local mapping for C-i. If both mappings are global then they are distinguished, if both mappings are local then they are distinguished. But if one is buffer local then both go to that one.

This is tricky. See #28022.
